### ISSUE TITLE
Docker: Environment Variables rather than hardcoded localhost

### ISSUE THREAD

**Issue by ncrmro** [mattkrick/meatier#52]
Thanks the project is awesome. Removed a big headache switching off meteor.

This takes a DATABASE_URL snippit variable below works. The issue is the graphql endpoint is hardcoded with localhost throught out as well so I'm thinking about having one env variable that replaces localhost how ever it's formatted throughout the project.

I've attempted to document this incase you'd like to have a docker branch.

```
export default {
  host : process.env.NODE_ENV === 'development' ? 'localhost' : process.env.DATABASE_URL ,
  port : 28015 ,
  db : process.env.NODE_ENV === 'testing' ? 'ava' : 'meatier' ,
  min : process.env.NODE_ENV === 'production' ? 50 : 3 ,
  buffer : process.env.NODE_ENV === 'production' ? 50 : 3
}
```

This is nifty as when you use docker You can make the DATABASE_URL inserted based on the docker container IP.

Dockerfile for those interested
```
FROM node:5

# use changes to package.json to force Docker not to use the cache
# when we change our application's nodejs dependencies:
ADD package.json /tmp/package.json
RUN cd /tmp && npm install
RUN mkdir -p /opt/app && cp -a /tmp/node_modules /opt/app/

# From here we load our application's code in, therefore the previous docker
# "layer" thats been cached will be used if possible
WORKDIR /opt/app
ADD . /opt/app

RUN npm run build

ENV DATABASE_HOST=localhost \
    DATABASE_PORT=28015 \
    GRAPHQL_HOST=localhost \
    GRAPHQL_PROTOCOL=http

EXPOSE 3000

ENTRYPOINT ["npm"]

CMD ["run", "prod"]
```
docker-compose.yml
```
version: '2'

services:
  db:
    image: rethinkdb
    ports:
        - "8080:8080"
        - "28015"
        - "29015"
    volumes:
        - /data

  app:
    build: .
    volumes:
      - ./src/:/opt/app/src
    ports:
        - "80:3000"
    command: start
    environment:
      DATABASE_HOST: meatier_db_1
      GRAPHQL_HOST: meatier_app_1
      GRAPHQL_PROTOCOL: http
```
tutum-stack.yml
```
lb:
  image: 'tutum/haproxy:latest'
  links:
    - web
  ports:
    - '80:80'
  roles:
    - global
  tags:
    - develop
    - meatier
rethinkdb:
  image: 'rethinkdb:latest'
  ports:
    - '8080:8080'
  tags:
    - develop
    - meatier
web:
  image: 'ncrmro/meatier:develop'
  environment:
    - DATABASE_HOST=rethinkdb
    - GRAPHQL_HOST=dev.meatier.com
  links:
    - rethinkdb
  tags:
    - develop
    - meatier
```

`docker build -t ncrmro/meatier .`

`docker tag -f ncrmro/meatier ncrmro/meatier:meatier`

`docker login --username=yourhubusername --email=youremail@company.com`

`docker push ncrmro/meatier:meatier`

**Comment by mattkrick** [comment]
@ncrmro i love where this is headed! Agreed the goal is to get as close as we can to a 1-click containerized dev environment. We're in the process of dockerizing the whole thing using azk. See: https://github.com/Araphel/meatier

For more info on the devops give a shoutout to @bartekus.

**Comment by ncrmro** [comment]
I've got this working on my end using the changes from Araphel/meatier.

**Comment by Bartekus** [comment]
Thank you and please feel free to resubmit any improvement onto the Araphel/meatier branch and I'll have them integrated and tested @ncrmro!

**Comment by ncrmro** [comment]
@Bartekus just saw this, will do. Azk really does work quite nicely.

**Comment by Bartekus** [comment]
Here is the fresh version of r3stack (rebrand of meatier wrapped in azk.io, but still the same meatier)
Have fun and if you see anything that can be improved, please submit a pull... I'll try to upgrade redux-simple-router to the new version react-router-redux v.4 tomorrow, but that's the last dependency that needs updating, I'll submit a update pull to Meatier tomorrow as well, so whichever way you prefer, meatier or r3stack, enjoy! Cheers!

**Comment by ncrmro** [comment]
Issue is resolved with #64
